src/components/ComparatorChangelogResults.tsx

```diff
@@ -0,0 +1,119 @@
+import {
+	Alert,
+	AlertIcon,
+	Box,
+	Heading,
+	Skeleton,
+	Stack,
+} from '@chakra-ui/react'
+
+import ProcessedReleaseChangeDescription from '~/components/ProcessedReleaseChangeDescription'
+import TextSkeleton from '~/components/TextSkeleton'
+import useProcessReleases from '~/hooks/useProcessReleases'
+import type { Release } from '~/models'
+import type { ProcessedRelease, ReleaseGroup, Repository } from '~/models'
+import { compareReleaseGroupsByPriority } from '~/utils'
+
+const ReleaseChangelogGroup = ({
+	title,
+	releaseGroup,
+	repository,
+	shouldShowTitle,
+}: {
+	title: ReleaseGroup
+	releaseGroup: Array<ProcessedRelease>
+	repository: Repository
+	shouldShowTitle: boolean
+}) => {
+	const textTransform =
+		title === 'breaking changes' ? 'uppercase' : 'capitalize'
+
+	return (
+		<Box key={title}>
+			{shouldShowTitle && (
+				<Heading
+					as="h2"
+					size="xl"
+					bgColor="background3"
+					mb={4}
+					py={4}
+					textTransform={textTransform}
+					position="sticky"
+					top={0}
+				>
+					{title}
+				</Heading>
+			)}
+			<Box mb={4}>
+				{releaseGroup.map((processedReleaseChange: ProcessedRelease) => (
+					<ProcessedReleaseChangeDescription
+						key={`${title}-${processedReleaseChange.id}`}
+						repository={repository}
+						processedReleaseChange={processedReleaseChange}
+						mb={8}
+					/>
+				))}
+			</Box>
+		</Box>
+	)
+}
+
+interface ComparatorChangelogResultsProps {
+	releases: Array<Release> | null
+	repository: Repository
+}
+
+const ComparatorChangelogResults = ({
```

Comment:
This component has been extracted from `RepositoryReleasesChangelog.tsx`. With this improvement, I have more control over the workflow of rendering results, errors or loading states.